src/login.js

```diff
@@ -23,23 +24,59 @@ let providers = {
 class Login extends Component {
   constructor(props) {
     super(props);
+    
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    
+    this.state = {
+      username: '',
+      password: ''
+    }
   }
-
-  renderUserCredentialsEnabled() {
-    if (this.props.userCredentialsEnabled) {
+  
+  handleChange(e) {
+    const target = e.target;
+    const name = target.name;
+    this.setState({
+      [name]: target.value
+    });
+  }
+  
+  handleSubmit(e){
+    e.preventDefault();
+    const options = {
+      uri: this.props.apiAuthorization,
```

Comment:
Please rename to authUrl

Reply:
uri is the parameter of request-promise, I don't think we can rename it.

Reply:
I meant the props key :). So: this.props.authUrl

Also with password grant type we use token endpoint (in props we will have both authUrl and tokenUrl, but password use only tokenUrl)

Reply:
Also the data is sent by POST method